pkg/user/repository.go

```diff
@@ -45,28 +38,29 @@ func (r userRepository) Create(u *model.User) error {
 	return err
 }
 
-func (r userRepository) FindByEmail(email string) (*model.User, error) {
+func (r repository) findByEmail(email string) (*model.User, error) {
 	var u *model.User
 	err := r.db.Where("email = ?", email).First(&u).Error
 	return u, err
 }
 
-func (r userRepository) FindOrCreate(user *model.User) (*model.User, error) {
+func (r repository) findOrCreate(user *model.User) (*model.User, error) {
 	var u *model.User
 	err := r.db.Where(model.User{Email: user.Email}).Attrs(model.User{Password: user.Password}).FirstOrCreate(&u).Error
 	return u, err
 }
 
-func (r userRepository) FindById(id uint) (*model.User, error) {
+func (r repository) FindById(id uint) (*model.User, error) {
 	var u *model.User
 	err := r.db.
 		Preload("Groups").
 		First(&u, id).Error
 	if err != nil {
 		if errors.Is(err, gorm.ErrRecordNotFound) {
-			return u, errdef.NotFound(fmt.Errorf("failed to find user with id %d: %v", id, err))
+			err := fmt.Errorf("failed to find user with id %d: %v", id, err)
```

Comment:
```suggestion
			err := fmt.Errorf("failed to find user with id %d: %s", id, err)
```

We want the error string of `err` in the error we return. My mistake 😋 I like using `%v` with `Printf` as it picks the default verb for me depending on the type.  Here its not what I want 😬